### PR TITLE
Fix filenames

### PR DESCRIPTION
**Pull request by alesgenova**
Remove fileName / fileNames redundancy in state files

### CODE REVIEW

File: tomviz/DataSource.cxx [hunk 1]
```diff
@@ -232,16 +232,26 @@ bool DataSource::appendSlice(vtkImageData* slice)
 
 void DataSource::setFileName(const QString& filename)
 {
-  auto reader = m_json.value("reader").toObject(QJsonObject());
-  reader["fileName"] = filename;
-  m_json["reader"] = reader;
+  // AG: remove fileName and fileNames redundancy. but keep backward
```
Comment (cryos) [inline tomviz/DataSource.cxx]: We can see from git blame who authored comments, no need to prefix with initials etc. Ideally keep as a sentence, capital first letter, full stop/period. I think this should be removed (looks more like the commit message to me), or reworded to explain that it is a convenience method.

File: tomviz/DataSource.cxx [hunk 2]
```diff
@@ -232,16 +232,26 @@ bool DataSource::appendSlice(vtkImageData* slice)
...
+  // compatibility.
+  QStringList fileNames = QStringList(filename);
+  setFileNames(fileNames);
+
+  // auto reader = m_json.value("reader").toObject(QJsonObject());
```
Comment (cryos) [inline tomviz/DataSource.cxx]: Should just remove code.

File: tomviz/DataSource.cxx [hunk 3]
```diff
@@ -232,16 +232,26 @@ bool DataSource::appendSlice(vtkImageData* slice)
...
+  // reader["fileName"] = filename;
+  // m_json["reader"] = reader;
 }
 
 QString DataSource::fileName() const
 {
+  // AG: remove fileName and fileNames reduntancy. but keep backward
```
Comment (cryos) [inline tomviz/DataSource.cxx]: As above, change looks reasonable but commit message would be better place to explain the change.

File: tomviz/LoadDataReaction.cxx [hunk 4]
```diff
@@ -168,16 +168,16 @@ DataSource* LoadDataReaction::loadData(const QStringList& fileNames,
   bool child = options["child"].toBool(false);
 
   DataSource* dataSource(nullptr);
-  QString fileName;
+  QString firstFile;
```
Comment (cryos) [inline tomviz/LoadDataReaction.cxx]: This feels odd, not sure if it is necessary but if you and @cjh1 want it for additional clarity it is fine by me.

File: tomviz/LoadDataReaction.cxx [hunk 5]
```diff
@@ -246,10 +246,9 @@ DataSource* LoadDataReaction::loadData(const QStringList& fileNames,
   }
 
   // Now for house keeping, registering elements, etc.
-  dataSource->setFileName(fileName);
-  if (fileNames.size() > 1) {
-    dataSource->setFileNames(fileNames);
-  }
+  // always save it as a list, even if there is only one file.
```
Comment (cryos) [inline tomviz/LoadDataReaction.cxx]: Capitalize first letter of first work.

File: .gitignore [hunk 6]
```diff
@@ -1,3 +1,4 @@
 *.pyc
 .DS_Store
 .cache
+.vscode
```
Comment (cryos) [inline .gitignore]: We don't add editor specific files to this, you can set up a global ignore locally.

**Comment by cjh1** [comment]
@cryos Are you happy for this to go in now? I think @alesgenova has addressed the review comments?

**Comment by cjh1** [comment]
Actually, looks like you wanted the commit messages reworked?

**Comment by cryos** [comment]
It is fine, I just reverted the variable name change, and pushed that, I think we can merge now.

**Pull request merged**